java/vector/src/main/java/org/apache/arrow/vector/complex/impl/ComplexWriterImpl.java

```diff
@@ -159,6 +164,11 @@ public MapWriter rootAsMap() {
     return mapRoot;
   }
 
+  @Override
+  public MapWriter rootAsMap() {
+    return rootAsMap(null);
```

Comment:
there are more than two binary situations here, using Boolean and null is probably not the best solution. What would you suggest?
1. init call, caseSensitive = null -> case insensitive
2. init call, caseSensitive = true/false -> case sensitive/insensitive
3. non-init call, caseSensitive = null -> already initialized, doing nothing
4. non-init call, caseSensitive = true/false -> if not the same with initialized sensitivity, IllegalArgumentException is thrown, otherwise, doing nothing